File: app/src/main/java/com/example/dkalev/remember/model/CardDao.java
package com.example.dkalev.remember.model;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import io.reactivex.Flowable;

@Dao
public interface CardDao {

    @Insert
    void insertAll(Card... cards);

    @Query("SELECT * FROM card WHERE :cardId == uid")
    Flowable<Card> getCard(int cardId);

    @Update
    void updateCard(Card card);


    @Query("DELETE FROM card WHERE :cardId == uid")
    void deleteCard(int cardId);
}

File: app/src/main/java/com/example/dkalev/remember/model/card/CardDao.kt
package com.example.dkalev.remember.model.card

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.Query
import android.arch.persistence.room.Update
import io.reactivex.Flowable

@Dao
interface CardDao {

    @Insert
    fun insertAll(vararg cards: Card)

    @Query("SELECT * FROM card WHERE :cardId == uid")
    fun getCard(cardId: Int): Flowable<Card>

    @Update
    fun updateCard(card: Card)


    @Query("DELETE FROM card WHERE :cardId == uid")
    fun deleteCard(cardId: Int)
}